### PR TITLE
Add Username and Logout Link to FlowV2

### PR DESCRIPTION
**Pull request by marcohald**
As stated here https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html the Webview should be a one time view.

> This should be a one time webview. Which means:
> 
>     There should be no cookies set when creating the webview
>     Passwords should not be stored
>     No state should be preserved after the webview has terminated
This is simply not correct in a few Applications for example the Enpass iOS App and the Nextcloud Desktop Client which opens the default Browser on Windows.
On Windows you can simply navigate in your Browser and logout and use another Account.
In the iOS Webview you cannot navigate away from the page and so you can't logout to use another Account.

I added the Username and the Logout URL to the Flow V2 which look now like this:
grafik
Can somebody give me a hint how to change the translation correct ?
I found this file for example: https://github.com/nextcloud/server/blob/310e6550b1c71b1415c6e7e25b892eebe231139f/core/l10n/de.json#L328 Do I need to add this manual to every language file ?

### CODE REVIEW

File: core/templates/loginflowv2/grant.php [hunk 1]
```diff
@@ -30,10 +30,16 @@
...
 	<p class="info">
-		<?php print_unescaped($l->t('You are about to grant %1$s access to your %2$s account.', [
+		<?php print_unescaped($l->t('You are about to grant %1$s access to your %2$s %3$s account.', [
 			'<strong>' . \OCP\Util::sanitizeHTML($_['client']) . '</strong>',
+			'<strong>' . \OCP\Util::sanitizeHTML(strval(\OC_User::getUser())) . '</strong>',
 			\OCP\Util::sanitizeHTML($_['instanceName'])
-		])) ?>
+		]));
+			print_unescaped('<br/>');
+			print_unescaped($l->t('Click %1$shere%2$s to logout.', [
```
Comment (LukasReschke) [inline core/templates/loginflowv2/grant.php:39]: I'm not a big fan of adding yet thing to this UI that already is confusing enough.

In fact, I'd argue that [the documentation here is incorrect](https://docs.nextcloud.com/server/latest/developer_manual/client_apis/LoginFlow/index.html#opening-the-webview):

> There should be no cookies set when creating the webview

The ability to re-use a cookie here allows for a better user experience. 

The fact that the users here can't be switched seems indeed like something to improve though. I'd ideally prefer a better UI experience though, take a look how Google does it:

account-verification

"Logout" is a very confusing term that most non-technical people would probably not directly understand in this context. If we however:

1. Highlight the current user better (e.g. a profile picture and the name)
2. Add a button "I dont recognize this account" / "Use another account" or something like that

This whole UX would be better here.

**Comment by LukasReschke** [comment]
> Do I need to add this manual to every language file ?

No need to. We have automation in place that does this for every merged commit automatically.

File: core/templates/loginflowv2/grant.php (hunk 1 above)
Comment (marcohald) [reply core/templates/loginflowv2/grant.php]: The Avatar is a great Idea, however I'm struggling with getting the userid to get the avatar URL.
I think i didnt get it how the templating work
I tried to use the code from https://github.com/nextcloud/server/blob/3a30ac495bd9663ef8233aacc8ce5b21a8adce67/lib/private/TemplateLayout.php#L129 
but i don't know why it is null everytime
grafik

File: core/Controller/ClientFlowLoginV2Controller.php [hunk 2]
```diff
@@ -62,6 +64,14 @@ class ClientFlowLoginV2Controller extends Controller {
 	private $userId;
 	/** @var IL10N */
 	private $l10n;
+	/** @var serverContainer */
```
Comment (ChristophWurst) [inline core/Controller/ClientFlowLoginV2Controller.php:67]:
```suggestion
	/** @var IServerContainer */
```

Comment (ChristophWurst) [reply core/Controller/ClientFlowLoginV2Controller.php:67]: but you actually never use the ``$this->serverContainer`` memeber, do you?

**Comment by skjnldsv** [comment]
As there is no feedback since a while I will close this ticket. 
If you will decide to work on this feature again and if it hasn't been fixed or implemented already, feel free to re-open and solve the various conflicts.

Thanks for the interest in Nextcloud and the effort put into this! 🙇

**Pull request closed without merging**